app/services/induction/find_relevant_to.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Induction::FindRelevantTo < BaseService
+  def call
+    query = participant_profile.induction_records
+
+    query = add_provider_to(query:) if lead_provider.present? || delivery_partner.present?
+    query = add_schedule_to(query:) if schedule.present?
+    query = add_school_to(query:) if school.present?
+    query = add_date_range_to(query:) if date_range.present?
+
+    query = query.current if current_not_latest_record
+    query.latest
+  end
+
+private
+
+  attr_reader :participant_profile, :lead_provider, :delivery_partner, :schedule, :school, :date_range, :current_not_latest_record
+
+  def initialize(participant_profile:, lead_provider: nil, delivery_partner: nil, schedule: nil, school: nil, date_range: nil, current_not_latest_record: false)
```

Comment:
This is great, knowing you at least need a `participant_profile` but can hone the results further by adding more criteria makes total sense.